src/charm.py

```diff
@@ -116,6 +109,36 @@ def _configure_pebble_layer(
         self._container.replan()
         self.unit.status = ActiveStatus()
 
+    def _on_fiveg_nrf_relation_joined(self, event: RelationJoinedEvent) -> None:
+        """Handle fiveg-nrf relation joined event.
+
+        Args:
+            event: RelationJoinedEvent
+        """
+        if not self.unit.is_leader():
+            return
+        if not self._nrf_service_is_running:
+            event.defer()
+            return
+        print(self._nrf_service_is_running)
+        self.nrf_provider.set_nrf_information(
+            url=NRF_URL,
+            relation_id=event.relation.id,
+        )
+
+    @property
+    def _config_file_is_written(self) -> bool:
+        """Returns whether the config file was written to the workload container.
+
+        Returns:
+            bool: Whether the config file was written.
+        """
+        if not self._container.exists(f"{BASE_CONFIG_PATH}/{CONFIG_FILE_NAME}"):
+            logger.info(f"Config file is not written: {CONFIG_FILE_NAME}")
+            return False
+        logger.info("Config file is written")
```

Comment:
I think there's no need to log here either. If this function returns `False`, charm will go to Waiting status which is enough I guess.

Reply:
As above, in the [opened draft pr](https://github.com/canonical/sdcore-nrf-operator/pull/13) I will be completely removing this function :)

Reply:
Nope, that's fine.